azerothcore/azerothcore-wotlk
# Fix(Core/BWL): Suppression traps can be disarmed

## Changes Proposed:
-  Move the method outside the AI, where it should be.
-  Added a new action to distinguish the boss action and the disarm trap action and their behavior.
-  Fix an issue with the trap not being visible after it goes active again. (me->Respawn() just made it work)

## Issues Addressed:

- Closes #10155

## SOURCE:


## Tests Performed:

- Tested in-game

## How to Test the Changes:


1. Create a rogue and learn disarm trap.
2. .go gameobject 75133
3. Disarm it and wait until it respawns.
4. Try to disarm it again, it should work.

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
